tests/storage/test_extraction.py

```diff
@@ -20,4 +34,294 @@
     ],
 )
 def test_prepare_x_axis(x_axis, expected):
-    assert expected == _prepare_x_axis(x_axis)
+    assert expected == extraction._prepare_x_axis(x_axis)
+
+
+class ErtConfigBuilder:
+    def __init__(self):
+        self.ensemble_size = 1
+        self._priors = {}
+
+    def add_prior(self, name, entry):
+        assert name not in self._priors
+        self._priors[name] = entry
+        return self
+
+    def build(self, path=None):
+        if path is None:
+            path = Path.cwd()
+        (path / "JOB").write_text("EXECUTABLE /bin/true\n")
+
+        self._build_ert(path)
+        self._build_priors(path)
+
+        config = ResConfig(str(path / "test.ert"))
+        enkfmain = EnKFMain(config)
+
+        # The C code doesn't do resource counting correctly, so we need to hook
+        # ResConfig to EnKFMain because otherwise ResConfig will be deleted and
+        # EnKFMain will use a dangling pointer.
+        enkfmain.__config = config
+
+        return LibresFacade(enkfmain)
+
+    def _build_ert(self, path):
+        f = (path / "test.ert").open("w")
+
+        # Default
+        f.write(
+            "JOBNAME poly_%d\n"
+            f"NUM_REALIZATIONS {self.ensemble_size}\n"
+            "INSTALL_JOB job JOB\n"
+            "SIMULATION_JOB job\n"
+        )
+
+    def _build_priors(self, path):
+        if not self._priors:
+            return
+
+        with (path / "test.ert").open("a") as f:
+            f.write("GEN_KW COEFFS coeffs.json.in coeffs.json coeffs_priors\n")
+        with (path / "coeffs.json.in").open("w") as f:
+            f.write("{\n")
+            f.write(",\n".join(f'  "{name}": <{name}>' for name in self._priors))
+            f.write("}\n")
+        with (path / "coeffs_priors").open("w") as f:
+            for name, entry in self._priors.items():
+                f.write(f"{name} {entry}\n")
+
+
+@pytest.fixture(autouse=True)
+def _chdir_tmp_path(monkeypatch, tmp_path):
+    """
+    All tests in this file must be run in a clean directory
+    """
+    monkeypatch.chdir(tmp_path)
+
+
+def test_empty_ensemble(client):
```

Comment:
Was just wondering if we are to label  `test_extraction.py` as `storage/integration` or it's `integration` be default.